src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.Netstandard2_0.cs

```diff
@@ -0,0 +1,841 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.DependencyModel
+{
+    public class DependencyContextJsonReader : IDependencyContextReader
```

Comment:
In general, it feels like there is too much duplication in this file. Is there a way to factor this out so the entire DependencyContextJsonReader isn't duplicated? Similar to the way we have "pal" layers in corefx - have a common DependencyContextJsonReader.cs file that contains all the common code, and then partial classes for all the "TFM-specific" code.